Cython/Includes/cpython/datetime.pxd

```diff
@@ -89,9 +150,22 @@ cdef extern from "datetime.h":
         object (*Time_FromTime)(int, int, int, int, object, PyTypeObject*)
         object (*Delta_FromDelta)(int, int, int, int, PyTypeObject*)
 
+        # We cannot use the following because we still need to compile on previous versions,
+        # instead we use datetime.h's macros that we can do some C macro magic on top of.
+
+        # constructors
+        # object (*TimeZone_FromTimeZone)(PyObject *offset, PyObject *name)
+
+        # singletons
+        # PyObject *TimeZone_UTC
```

Comment:
```suggestion
        # We cannot use the following because they do not compile in Python < 3.7.
        # Instead, we use datetime.h's macros here that we can backport in C.

        # constructors
        object (*TimeZone_FromTimeZone)(object offset, PyObject *name)

        # singletons
        PyObject *TimeZone_UTC
```

Reply:
```diff
-    return <object>PyDateTime_TimeZone_UTC
+    if PY_VERSION_HEX < 0x030700b1:
+        # TODO: emulate
+        raise RuntimeError("Time zones are not available from the C-API")
+    return PyDateTime_TimeZone_UTC
```
Referencing this suggestion, I tried something similar earlier but it doesn't work. I get:

```
------------------------------------------------------------
...
# This is because of immutable nature of these objects by design.
# If you would change time/date/datetime/timedelta object you need to recreate.

# Get UTC singleton
cdef inline object get_utc():
    if PY_VERSION_HEX < 0x030700b1:
      ^
------------------------------------------------------------

datey.pxd:279:7: undeclared name not builtin: PY_VERSION_HEX
```

Reply:
`from cpython.version cimport PY_VERSION_HEX`

Reply:
I was able to do this, not sure if there is a better way (if so please point that out):
```py
cdef inline object get_utc():
    cast = <object>PyDateTime_TimeZone_UTC
    if cast is None:
        raise RuntimeError('Time zones are not available from the C-API.')
    # TODO: Emulate the UTC singleton
    return cast
```

Reply:
I tried my best to look into emulating the UTC timezone, but without being able to create timezones (which is not exposed to us) that is impossible.